source/controllers/PPGameStateController.cpp

```diff
@@ -22,32 +22,48 @@ void GameStateController::_jsonv1_loadQueues(const json_t &queues) {
     _state.wrongActions.clear();
     _state.recorded.clear();
     _state.obstacles.clear(); 
+    _state.healthPotions.clear(); 
+    _state.healthBack = 0; 
     // Build each queue.
     for (const auto &queue : queues->asArray()) {
         vec<vec<uint>> queue_s;
         vec<bool> wa_queue_s;
         vec<bool> r_queue_s;
         vec<bool> obs_queue_s;
+        vec<bool> hlth_queue_s;
         // Build canvas of each queue.
         for (const auto &canvas : queue->asArray()) {
             const auto r = canvas->asIntArray();
             // This is to cast vec<int> to vec<uint>.
             vec<uint> colors(r.begin(), r.end());
+            //Bomb obstacle
             if (colors[colors.size() - 1] == ((uint)10)) {
                 obs_queue_s.push_back(true);
+                hlth_queue_s.push_back(false);
+               hlth_queue_s.push_back(false);
                 colors.pop_back();
             }
+            //Health Potion
+           else if (colors[colors.size() - 1] == ((uint)11)) {
```

Comment:
would it be better to have one common matrix for obstacles and health portions, and make them int arrays instead of bool arrays? Just for the sake of using fewer data structures for state management. It could be as simple as using 1 to indicate obstacles, 2 for health potions, 3 for beach balls (if that gets integrated), and -1 for 'normal' canvases. This is just a nit, and we can probably fix that later but if you have time this may be worth thinking about?

Reply:
I had considered that. At first it was mainly to keep logic for checking way simpler & less refactoring for obstacles but in hindsight it shouldn't be too much of a change to fix. I can try to fix that now